rasa_core/tracker_store.py

```diff
@@ -219,7 +242,14 @@ def save(self, tracker, timeout=None):
             {"$set": state},
             upsert=True)
 
-    def retrieve(self, sender_id):
+    def _metadata_matches(self, stored, metadata):
+        if not metadata:
+            return True
+        return stored.get("metadata") == metadata
+
+    def retrieve(self, sender_id: Text,
```

Comment:
why do we need the metadata matching - couldn't the one requesting the data from core decide the correct instance to post the request to?